sklearn/inspection/_plot/decision_boundary.py

```diff
@@ -164,16 +164,36 @@ class DecisionBoundaryDisplay:
     """
 
     def __init__(
-        self, *, xx0, xx1, response, multiclass_colors=None, xlabel=None, ylabel=None
+        self,
+        *,
+        xx0,
+        xx1,
+        response,
+        multiclass_colors=None,
+        xlabel=None,
+        ylabel=None,
+        response_method_used=None,
+        estimator_name=None,
     ):
         self.xx0 = xx0
         self.xx1 = xx1
         self.response = response
         self.multiclass_colors = multiclass_colors
         self.xlabel = xlabel
         self.ylabel = ylabel
+        self.response_method_used_ = response_method_used
+        self.estimator_name_ = estimator_name
```

Comment:
I'd move this up, under `multiclass_colors` (also makes it more consistent with order in other displays: #30717)